diff --git a/src/AppBundle/Security/JwtTokenAuthenticator.php b/src/AppBundle/Security/JwtTokenAuthenticator.php
index fc6ccf1..1bf8999 100644
--- a/src/AppBundle/Security/JwtTokenAuthenticator.php
+++ b/src/AppBundle/Security/JwtTokenAuthenticator.php
@@ -49,9 +49,17 @@ class JwtTokenAuthenticator extends AbstractGuardAuthenticator
 
         $username = $data['username'];
 
-        return $this->em
+        $user = $this->em
             ->getRepository('AppBundle:User')
             ->findOneBy(['username' => $username]);
+
+        if (!$user) {
+            throw new CustomUserMessageAuthenticationException(
+                'That API token is not normcore'
+            );
+        }
+
+        return $user;
     }
 
     public function checkCredentials($credentials, UserInterface $user)
