diff --git a/app/config/services.yml b/app/config/services.yml
index 5b2a6d9..86d0aa2 100644
--- a/app/config/services.yml
+++ b/app/config/services.yml
@@ -5,3 +5,5 @@ services:
 
     spooky_user_provider:
         class: AppBundle\Security\SpookyUserProvider
+        arguments:
+            - '@doctrine.orm.entity_manager'
diff --git a/load.php b/load.php
new file mode 100644
index 0000000..6b9c690
--- /dev/null
+++ b/load.php
@@ -0,0 +1,17 @@
+<?php
+
+use AppBundle\Entity\User;
+
+$loader = require __DIR__.'/app/autoload.php';
+
+$kernel = new AppKernel('dev', true);
+$kernel->boot();
+
+$container = $kernel->getContainer();
+
+$em = $container->get('doctrine.orm.entity_manager');
+
+$user = new User('weaverryan', 'ryan@knpuniversity.com');
+$em->persist($user);
+$em->flush();
+
diff --git a/src/AppBundle/Entity/User.php b/src/AppBundle/Entity/User.php
index 9073539..9f69100 100644
--- a/src/AppBundle/Entity/User.php
+++ b/src/AppBundle/Entity/User.php
@@ -4,14 +4,35 @@ namespace AppBundle\Entity;
 
 use Symfony\Component\Security\Core\Role\Role;
 use Symfony\Component\Security\Core\User\UserInterface;
+use Doctrine\ORM\Mapping as ORM;
 
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="user")
+ */
 class User implements UserInterface
 {
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string")
+     */
     private $username;
 
-    public function __construct($username = null)
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $email;
+
+    public function __construct($username = null, $email = null)
     {
         $this->username = $username;
+        $this->email = $email;
     }
 
     public function getUsername()
@@ -38,4 +59,4 @@ class User implements UserInterface
     {
         $this->username = $username;
     }
-}
\ No newline at end of file
+}
diff --git a/src/AppBundle/Security/FormLoginAuthenticator.php b/src/AppBundle/Security/FormLoginAuthenticator.php
index c1c3a57..a4c17ce 100644
--- a/src/AppBundle/Security/FormLoginAuthenticator.php
+++ b/src/AppBundle/Security/FormLoginAuthenticator.php
@@ -3,6 +3,7 @@
 namespace AppBundle\Security;
 
 use AppBundle\Entity\User;
+use Doctrine\ORM\EntityManager;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
@@ -19,9 +20,12 @@ class FormLoginAuthenticator extends AbstractFormLoginAuthenticator
 {
     private $router;
 
-    public function __construct(RouterInterface $router)
+    private $em;
+
+    public function __construct(RouterInterface $router, EntityManager $em)
     {
         $this->router = $router;
+        $this->em = $em;
     }
 
     public function getCredentials(Request $request)
@@ -40,10 +44,9 @@ class FormLoginAuthenticator extends AbstractFormLoginAuthenticator
     {
         $username = $credentials['username'];
 
-        $user = new User();
-        $user->setUsername($username);
-
-        return $user;
+        return $this->em
+            ->getRepository('AppBundle:User')
+            ->findOneBy(['username' => $username]);
     }
 
     public function checkCredentials($credentials, UserInterface $user)
diff --git a/src/AppBundle/Security/SpookyUserProvider.php b/src/AppBundle/Security/SpookyUserProvider.php
index 4f7f768..7eaf776 100644
--- a/src/AppBundle/Security/SpookyUserProvider.php
+++ b/src/AppBundle/Security/SpookyUserProvider.php
@@ -3,27 +3,40 @@
 namespace AppBundle\Security;
 
 use AppBundle\Entity\User;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
 use Symfony\Component\Security\Core\User\UserInterface;
 use Symfony\Component\Security\Core\User\UserProviderInterface;
 
 class SpookyUserProvider implements UserProviderInterface
 {
+    private $em;
+
+    public function __construct(EntityManager $em)
+    {
+        $this->em = $em;
+    }
+
     public function loadUserByUsername($username)
     {
-        // "load" the user - e.g. load from the db
-        $user = new User();
-        $user->setUsername($username);
+        $user = $this->em->getRepository('AppBundle:User')
+            ->findOneBy(['username' => $username]);
+
+        if (!$user) {
+            throw new UsernameNotFoundException();
+        }
 
         return $user;
     }
 
     public function refreshUser(UserInterface $user)
     {
-        return $user;
+        return $this->em->getRepository('AppBundle:User')
+            ->findOneBy(['username' => $user->getUsername()]);
     }
 
     public function supportsClass($class)
     {
         return $class == 'AppBundle\Entity\User';
     }
-}
\ No newline at end of file
+}
